Airport1/Airport1/Program.cs

```diff
@@ -0,0 +1,470 @@
+﻿using System;
+using System.Globalization;
+
+namespace Airport1
+{
+    public static class Program
+    {
+        public enum FlightsArrayLevel
+        {
+            FlightType,
+            DateTime,
+            CityPort,
+            Airline,
+            Terminal,
+            FlightStatus,
+            Gate
+        }
+
+        public static object[,] FlightsArray;
+        public static bool EmergencyIsSet;
+        public static EmergencyType EmergencyStatus;
+
+        static void Main(string[] args)
+        {
+            Console.WriteLine(DateTime.Now);
+
+            while (true)
+            {
+                if (EmergencyIsSet)
+                {
+                    Console.ForegroundColor = ConsoleColor.Red;
+                    Console.WriteLine($"Evacuation, reason: {EmergencyStatus}!!!");
+                    Console.ForegroundColor = ConsoleColor.White;
+                    return;
+                }
+
+                Console.WriteLine(@$"Please choose operation.
+{(int)OperationType.AddFlight} - Add flight
+{(int)OperationType.ShowArrival} - Arrivals
+{(int)OperationType.ShowDepartures} - Departures
+{(int)OperationType.EditInformation} - Edit inforamtion
+{(int)OperationType.SearchByFlightNumber} - Search by flight number
+{(int)OperationType.SearchByTime} - Search by specified time in airport
+{(int)OperationType.SearchNearest} - Search nearest
+{(int)OperationType.SearchByPort} - Search by port
+{(int)OperationType.SetEmergencyStatus} - Set emergency status
+{(int)OperationType.Exit} - Exit");
+
+                var operationType = (OperationType)int.Parse(Console.ReadLine());
+                switch (operationType)
+                {
+                    case OperationType.AddFlight: AddFlight(); break;
+                    case OperationType.ShowArrival: ShowArrivals(); break;
+                    case OperationType.ShowDepartures: ShowDepartures(); break;
+                    case OperationType.EditInformation: EditInformation(); break;
+                    case OperationType.SearchByFlightNumber: SearchByFlightNumber(); break;
+                    case OperationType.SearchByTime: SearchByTime(); break;
+                    case OperationType.SearchNearest: SearchNearest(); break;
+                    case OperationType.SearchByPort: SearchByPort(); break;
+                    case OperationType.SetEmergencyStatus: SetEmergencyStatus(); break;
+                    case OperationType.Exit: return;
+                    default: Console.WriteLine($"Program doesn't has operation type id {(int)operationType}"); break;
+                }
+            }
+        }
+
+        public static void AddFlight()
+        {
+            Console.WriteLine();
+            Console.WriteLine(@$"Enter flight type:
```

Comment:
Blocks line 70-106 and line 205-240 are equal. Maybe it would be effective to realise them as additional method?

Reply:
Lines 70-106 and 205-240 are equal, but different logic implemented, as you see on line 107 I add new flight to FlightsArray, AppendFlight requires not only one argument, as a result I must create method that will return much values. I can't create method that will return more than one value, using object array isn't the best option, IMHO